src/IterableCounter.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection;
+
+use function count;
+use function Pipeline\take;
+
+/**
+ * @internal
+ */
+final class IterableCounter
+{
+    /**
+     * @param iterable<mixed> $subjects
+     */
+    public static function bufferAndCountIfNeeded(iterable &$subjects, bool $runConcurrently): int
+    {
+        if ($runConcurrently) {
+            // This number is typically fed to ProgressFormatter/ProgressBar or variants.
+            // In progress bar lingo 0 stands for an unknown number of steps.
+            return 0;
```

Comment:
just an idea - probably it would be more clear to name it `UNKNON_COUNT` and use in ProgressFormatter/ProgressBar/etc this constant. Then it's even not necessary to be zero - any value

up to you

Reply:
Since we already nagged a lot in this PR though I think it would be better in a separate PR, especially since it may affect a few more files. But :+1: on a constant name there it makes it clearer

Reply:
That is an excellent idea.

Reply:
Implemented. It might be not the right place. Please tell me what you think.